packages/deployer/src/deploySolutionItems.ts

```diff
@@ -577,8 +590,12 @@ export function _createItemFromTemplateWhenReady(
   // ensure this is present
   template.dependencies = template.dependencies || [];
   // if there is no entry in the templateDictionary, add it
-  if (!templateDictionary.hasOwnProperty(template.itemId)) {
-    templateDictionary[template.itemId] = {};
+  if (
+    !templateDictionary.hasOwnProperty(template.itemId) ||
+    !getProp(templateDictionary[template.itemId], "def")
```

Comment:
"def" seems like a magic word--perhaps add an inline comment or reference about it?